### PR TITLE
fix(ui): revamp teams page added supported subscription webhooks

### LINKED ISSUE #12533: Improvements for team entity
**Is your feature request related to a problem? Please describe.**
1. Every team in our organization have slack channels for the team. 
It's used for automatic alerts, questions etc.
We need a place in OMD to associate the team slack channel.  

2. Inside the team, each user have a different role (BA, DE, Lead etc.).
If someone looks to ask someone from the team a question, he will need to know the role usually to choose the right person.
It may be connected to roles and permissions as well, but we are not there yet, and it's not simple as just the actual role in the team which have policies etc.

**Describe the solution you'd like**
1. Have a new section for communication channels. 
That include for example:
Channel type/description | channel link/name (for the case of slack)

3. Add 'job title' column here (probably better to have closed list to choose from somewhere?). 
image

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes #12533

- Revamp teams page 
- supported subscription webhook.
- accessiable for team owner and member and editable to team owner and on having edit access

image

image

image

image
image

image

https://github.com/open-metadata/OpenMetadata/assets/66266464/ef9e3b18-821e-485d-a8d2-e55d9f8d00e9

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

**Comment by harshach** [comment]
@chirag-madlani @Ashish8689 can we allow users to minimize the panel above for teams and user profiles. Helps them to view more of the teams/assets and users/feeds etc..

**Comment by harshach** [comment]
also on the team profile links about slack, 
1. We should allow users to configure only one type of Slack/MsTeams/GChat. Not sure there is a usecase to have multiple of these cc @mohityadav766 
2. These are incoming web hooks right, this is potentially secret information. Except for team members or team owners we shouldn't be showing this information

File: openmetadata-ui/src/main/resources/ui/src/constants/Teams.constants.ts [hunk 1]
```diff
@@ -20,3 +20,32 @@ export const TABLE_CONSTANTS = {
     row: DraggableBodyRow,
   },
 };
+
+export enum SUBSCRIPTION_WEBHOOK_LABEL {
+  msTeams = 'Ms Teams',
+  slack = 'Slack',
+  gChat = 'G Chat',
+  generic = 'Generic',
+}
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/constants/Teams.constants.ts:29]: Localisation?

File: openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/TeamDetailsV1.tsx [hunk 2]
```diff
@@ -709,604 +635,510 @@ const TeamDetailsV1 = ({
...
+              {createTeamPermission && (
+                <Button
+                  data-testid="add-team"
                   title={
-                    isGroupType
-                      ? t('message.group-team-type-change-message')
-                      : t('label.edit-entity', {
-                          entity: t('label.team-type'),
-                        })
+                    createTeamPermission
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/TeamDetailsV1.tsx:684]: We won't render button if there's no permission
We can avoid unwanted condition here

File: openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/TeamDetailsV1.tsx [hunk 3]
```diff
@@ -709,604 +635,510 @@ const TeamDetailsV1 = ({
...
-                    permission={entityPermissions}
-                    searchText={teamUsersSearchText}
-                    users={currentTeamUsers}
-                    onAddUser={handleAddUser}
-                    onChangePaging={teamUserPagingHandler}
-                    onRemoveUser={removeUserFromTeam}
-                    onSearchUsers={handleTeamUsersSearchAction}
-                  />
-                )}
+              <>
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Team/TeamDetails/TeamDetailsV1.tsx:1066]: I feel Tabs component should handle this, Why we want this logic can you check it again?

**Review by chirag-madlani — DISMISSED** [review]

**Review by harshach — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**